### DISCUSSION TITLE
Guidance on star exports from entrypoint.js

### DISCUSSION THREAD

**Discussion by icereval** [ojkelly/yarn.build#102] [Q&A]
**Is your feature request related to a problem? Please describe.**
Unable to load the `handler` function from `entrypoint.js` in AWS Lambda.

**Describe the solution you'd like**
When loading the `bundle.zip` into AWS Lambda and referencing its `handler` function via `bundle/entrypoint.js`, it would be ideal if the `entrypoint.js` was re-exporting module functions.

**Describe alternatives you've considered**
Leveraging typescript, we can convert a simple export within the existing project, and thus copy its transpiled `__exportStar` polyfill to a custom `app.js` (nearly equivalent to `entrypoint.js`).

https://github.com/ojkelly/yarn.build/blob/trunk/packages/plugins/plugin-build/src/commands/bundle/index.ts#L274-L284

`app.js`
```javascript
'use strict';
require('./.pnp.js').setup();
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
  if (k2 === undefined) k2 = k;
  Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
  if (k2 === undefined) k2 = k;
  o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
  for (var p in m) if (p !== 'default' && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, '__esModule', { value: true });
__exportStar(require(`./packages/${process.env.PACKAGE_NAME}/dist/index.js`), exports);
```

**Additional context**
This would depend on the project's `tsconfig.json` target polyfills, and would likely be best to perform this step in the `yarn bundle` command itself, with a call to `tsc` and an `entrypoint.ts` file.

**Comment by ojkelly** [comment]
Are we able to do this in a way that's generic to any js file?

If so, then this sounds great.

**Comment by jacobwgillespie** [comment]
Out of curiosity, does something like this work?

```javascript
"use strict";

const pnp = require("${pnp}").setup();

module.exports = require("./${main}");
```

That would effectively set the current module's exports equal to _all_ the exports of the main script - assuming that the main script itself set `__esModule`, I believe this means that `entrypoint.js` would as well? So perhaps this would just work? I don't think you necessarily need to set `entrypoint.js` as built ESM with dynamic bindings, I think you just need to proxy everything over to the built `main` file.

I may be missing something here though.
